policy/policy-authority_test.go

```diff
@@ -133,11 +133,83 @@ func TestWillingToIssue(t *testing.T) {
 	}
 }
 
-func TestChallengesFor(t *testing.T) {
+func TestCheckCAARecords(t *testing.T) {
+	shouldBeReserved := []string{
+		"google.com",
+		"mail.google.com",
+		"comodo.com",
+		"0day.net",
+		"darktangent.org",
+		"nails.eu.org",
+		"somaf.de",
+		"hackerjeopardy.com",
+		"instantssl.com",
+		"roe.ch",
+		"soulboxios.com",
+	}
+
+	shouldBeCritical := []string{
+		"www.zx.com",
+		"arrakis.tv",
+		"goop.org",
+		"mail2.bevenhall.se",
+		"matroxsolutions.com",
+		"nethemba.com",
+		"www.dotsport.info",
+		"dnsseczombo.com",
+		// these all seem to have NS records pointing to broken
+		// servers so may also throw CAAErrors....!
+		// "madtech.nl",
+		// "codewiz.org",
+		// "sugarlabs.org",
+		// "sunjammer.codewiz.org",
+		// "sunjammer.sugarlabs.org",
+		// "www.isoc.lu",
+	}
+
+	shouldBeAccepted := []string{
+		"bracewel.net",
+		"letsencrypt.org",
+		"linux.org",
+		"pir.org",
+	}
+
+	// shouldBeCAAError := []string {
+	//	"derrderrreeeerwerererwwww.moopa",
+	// }
+
 	// Audit logger
 	audit, _ := blog.Dial("", "", "tag")
 
-	pa := NewPolicyAuthorityImpl(audit)
+	pa := NewPolicyAuthorityImpl(audit, "letsencrypt.org")
+
+	for _, domain := range shouldBeReserved {
+		identifier := core.AcmeIdentifier{Type: core.IdentifierDNS, Value: domain}
+		if err := pa.CheckCAARecords(identifier); err != ReservedCAAError {
```

Comment:
It's good to have live tests for real-world sanity. It may introduce test flakiness if some of the relevant name servers are flaky. Ideally we'd also have fully local tests that mock out all network interaction. Is there a good way to do that?